applications/TopologyOptimizationApplication/custom_elements/small_displacement_simp_element.cpp

```diff
@@ -0,0 +1,302 @@
+        const SizeType strain_size = mConstitutiveLawVector[0]->GetStrainSize();
+
+        KinematicVariables this_kinematic_variables(strain_size, dimension, number_of_nodes);
+        ConstitutiveVariables this_constitutive_variables(strain_size);
+
+        //create constitutive law parameters:
+        ConstitutiveLaw::Parameters Values(GetGeometry(), GetProperties(), rCurrentProcessInfo);
+
+
+        //set constitutive law flags:
+        Flags &ConstitutiveLawOptions = Values.GetOptions();
+        ConstitutiveLawOptions.Set(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN, UseElementProvidedStrain());
+        ConstitutiveLawOptions.Set(ConstitutiveLaw::COMPUTE_STRESS, true);
+        ConstitutiveLawOptions.Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, false);
+
+        Values.SetStrainVector(this_constitutive_variables.StrainVector);
+
+        const std::size_t number_of_integration_points = integration_points.size();
+
+        
+        for (IndexType point_number = 0; point_number < number_of_integration_points; ++point_number) {
+                // Compute element kinematics B, F, DN_DX ...
+                CalculateKinematicVariables(this_kinematic_variables, point_number, this->GetIntegrationMethod());
+
+                // Compute material reponse
+                CalculateConstitutiveVariables(this_kinematic_variables, this_constitutive_variables, Values, point_number, integration_points, GetStressMeasure());
+
+                // Compute VM stress
+                if (dimension == 2 ) {
+                    rValues[point_number] = ConstitutiveLawUtilities<3>::CalculateVonMisesEquivalentStress(this_constitutive_variables.StressVector);
+                } else {
+                    rValues[point_number] = ConstitutiveLawUtilities<6>::CalculateVonMisesEquivalentStress(this_constitutive_variables.StressVector);
+                }
+        }
+	} 
+
+    // Additional part for post-processing of the topology optimized model part
+    else if (rVariable == X_PHYS)
+    {
+        for (SizeType PointNumber = 0; PointNumber < mConstitutiveLawVector.size(); PointNumber++)
+            rValues[PointNumber] = this->GetValue(X_PHYS);
+    }
+    else {
+
+            SmallDisplacement::CalculateOnIntegrationPoints(rVariable, rValues, rCurrentProcessInfo);
+      	}
+
+    KRATOS_CATCH( "" )
+    } 
+
+//************************************************************************************
+//************************************************************************************
+
+void SmallDisplacementSIMPElement::Calculate(const Variable<double> &rVariable, double &rOutput, const ProcessInfo &rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    if (rVariable == DCDX || rVariable == LOCAL_STRAIN_ENERGY)
+    {
+        // Get values
+        double E_min     = this->GetValue(YOUNGS_MODULUS_MIN);
+        double E_initial = this->GetValue(YOUNGS_MODULUS_0);
+        double E_current = this->GetValue(YOUNG_MODULUS);
```

Comment:
Not stored in properties?¿

Reply:
I am not sure if I can follow you?!

Reply:
YOUNG_MODULUS is stored in the element?, not in the properties?

Reply:
The memory usage will be huge

Reply:
When you mention it now, it makes sens! I kept it the way it was implemented, and in this way every element saves the young's modulus for itself (also the min young's modulus). How can I call the properties from the element? With the `rCurrentProcessInfo`?

Reply:
You need to differentiate between values that are the same for all elements (you can store them in the properties) and values that have different values for each element, those you have to store at the element. From what i understand, the current `YOUNG_MODULUS` is variing across elements, therefor has to be soterd at the element. 

Properties can be accessed like this: https://github.com/KratosMultiphysics/Kratos/blob/f55a2c57ac8102038bdb46428cb2d2a95c7b92cc/kratos/includes/element.h#L978

Reply:
I need to be clarified, there are more than one young modulus?

Reply:
Yes, there is a minimum Young's modulus, in order to avoid singularities in the stiffnessmatrix due to elements with a very low density. There is the material Young's modulus and there is the current Young's modulus, which is calulated with the density of the current element.
But as @armingeiser said, only the current Young's modulus has to be stored in the element, the others do not chage over the optimization process. 

My question is now, as the minimum Youngs modulus is defines in the optimization parameters, not in the mdpa file, how can I save this as a property of the optimization process?

Reply:
The ones that are constant for all elements should be stored in properties. These are stored in an independent json for materials and properties, check Structural tests for examples

Reply:
Is tehere also a possibility to store them in the properties in the optimization? For example the penalty therm `PENALTY`, which is also constant in the optimization could be saved there, but that is a optimization parameter and will be set in the optimization_parameters.json

Reply:
Yes you can, even if it is an optimization parameter

Reply:
How can I do that? Assuming I would like to store it in the properties in the topology_optimizer_factory.py?

Reply:
I assume you only have one property?, store values there, at least for the moment, and you can generalize in a future PR